### DISCUSSION TITLE
reCaptcha

### DISCUSSION THREAD

**Discussion by matteotestoni** [OpenMage/magento-lts#3891] [Ideas]
### Description (*)
in openmage lts 19 or 20 there are many spam messagges from newsletter subscribe and contacts

### Expected behavior (*)
captcha extension for openmage

### Benefits
reduce spam and increment security

### Additional information
maybe there are extension updated?

**Comment by fballiano** [comment]
this one https://github.com/empiricompany/reCaptcha

**Comment by addison74** [comment]
@fballiano - The captcha module in OpenMage is very old. The generated image is ancient like 16 years ago. We should remove it from OpenMage and move the code to a repository like we did for Backup and others.

**Reply by fballiano** [reply]
yeah, you know I'm all about removing old junk but at the moment there's only one solution, which is @empiricompany's recaptcha module, which depends on the mage_captcha if I'm not mistaken.
so.. I don't know :-\

**Comment by fballiano** [comment]
I started to work on a Cloudflare Turnstile module:
https://github.com/fballiano/openmage-cloudflare-turnstile

It already works although it supports only customer-registration, contact-form and newsletter-subscribe at the time of writing this comment.

It doesn't depend of Mage_Captcha for anything so Mage_Captcha could be disabled/removed (recaptha module depends on Mage_Captcha instead).

:-)
